bin/src/command/generate_command.dart

```diff
@@ -53,22 +54,30 @@ class GenerateCommand {
       ..writeln('}')
       ..writeln();
 
-    final dependencies = <Dependency, DependencyLock>{};
-    final sortedDependencies = params.dependencies..sort((a1, a2) => a1.name.compareTo(a2.name));
-    for (final dependency in sortedDependencies) {
-      dependencies[dependency] = params.pubspecLock.dependencies.firstWhere((element) => element.name == dependency.name);
+    final allDependencies = <Dependency>[];
+    allDependencies.addAll(params.extraDependencies);
+    for (final dependency in params.dependencies) {
+      if (allDependencies.where((element) => element.name == dependency.name).isEmpty) {
```

Comment:
This is something I want to update in the future because maybe we want to merge already added items together. Maybe you use a package that should also have a link to a specific website. But it not added to pub.dev you could use extra_licenses to add the homepage but the version could still be used from the pubspec.yaml